src/js/components/homepage/visualizations/treemap/TreeMap.jsx

```diff
@@ -67,27 +78,41 @@ export default class TreeMap extends React.Component {
                 windowWidth,
                 visualizationWidth: this.sectionWrapper.offsetWidth
             });
+            let colors = this.props.colors;
+            if (this.state.showSub) {
+                colors = this.props.alternateColors;
+            }
             if (this.props.categories.children.length > 0) {
-                this.buildTree(this.props.categories, this.props.colors, '');
+                this.buildTree(this.props.categories, colors, null, this.state.showSub);
             }
         }
     }
 
-    buildTree(cats, colors, chosen) {
+    buildTree(cats, colors, chosen, sub) {
         // put the data through d3's hierarchy system to sum and sort it
         const root = d3.hierarchy(cats)
         .sum((d) => (d.value))
         .sort((a, b) => b.height - a.height || b.value - a.value);
 
         // set up a treemap object and pass in the root
         let tileStyle = d3.treemapBinary;
+        let mapHeight = 565;
+        let mapWidth = this.state.visualizationWidth;
         if (this.state.windowWidth < 768) {
             tileStyle = d3.treemapSlice;
         }
+        if (sub === true) {
```

Comment:
The collapsed tree view in subtree mode should probably be its own component rather than a parameter of the normal treemap. Keeping it all as one monolith decreases the understandability and maintainability of the code.